shared-libs/transitions/src/transitions/registration.js

```diff
@@ -185,6 +186,27 @@ const setBirthDate = doc => {
   }
 };
 
+const getPhoneNumber = doc => {
+  if (!doc || !doc.fields) {
+    return '';
+  }
+  return doc.fields.phone_number;
+};
+
+const setPhoneNumber = doc => {
+  const phoneNumber = getPhoneNumber(doc);
+  // By default for a valid phone number SmsParser itself adds a country code if not provided
+  // So if country code is not present in phone_number throw error
+  const validPhone = phoneNumberParser.validate(config.getAll(), phoneNumber);
```

Comment:
Also, what if we don't want to allow duplicate phone numbers (as in contact forms)? What if others want to allow duplicates? Can that be made configurable?

Reply:
Now its configurable. Added a validation uniquePhone for the phone_number field which controls this.